libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -3154,6 +3154,16 @@ bool IRGeneratorForStatements::visit(TryCatchClause const& _clause)
 void IRGeneratorForStatements::setLocation(ASTNode const& _node)
 {
 	m_currentLocation = _node.location();
+
+	if (m_currentLocation.isValid())
+		m_code <<
+			"/// @origin " <<
+			m_currentLocation.source->name() <<
+			":" <<
+			m_currentLocation.start <<
```

Comment:
I'm a bit uneasy about just shifting numbers into the stream - there is just so much state that can affect how they are encoded. What about using `to_string` and string concatenation instead?